### PR TITLE
ncm-metaconfig: Include libvirtd conf file for the QEMU driver

### LINKED ISSUE #805: ncm-metaconfig: include libvirtd configuration file for the QEMU driver
`/etc/libvirtd/qemu.conf` is required in some specific configurations. As example to be able to use `virsh` command as non-root user or to provide QEMU access to the VFIO device files (to enable GPU passthrough for example).

### PR DESCRIPTION
**Pull request by alvarosimon**
* Fix #805 :  Include libvirtd configuration file for the QEMU driver

### CODE REVIEW

**Comment by stdweird** [comment]
LGTM

File: ncm-metaconfig/src/main/metaconfig/libvirtd/pan/schema.pan [hunk 1]
```diff
@@ -72,6 +72,63 @@ type type_libvirtd_audit = {
...
+
+type type_qemu_cgroup = {
+    'cgroup_controllers' ? string[]
+    'cgroup_device_acl' ? string[]
+};
+
+type type_qemu_image_format = {
+    'save_image_format' ? string with match(SELF, '^(raw|lzop|gzip|bzip2|xz)$')
+    'dump_image_format' ? string with match(SELF, '^(raw|lzop|gzip|bzip2|xz)$')
+    'snapshot_image_format' ? string with match(SELF, '^(raw|lzop|gzip|bzip2|xz)$')
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/libvirtd/pan/schema.pan]: You've used this multiple times, make a type for it!

File: ncm-metaconfig/src/main/metaconfig/libvirtd/pan/schema.pan [hunk 2]
```diff
@@ -72,6 +72,63 @@ type type_libvirtd_audit = {
...
+    'spice_listen' ? type_ip
+    'spice_tls' ? boolean
+    'spice_tls_x509_cert_dir' ? string
+    'spice_password' ? string
+    'spice_sasl' ? boolean
+    'spice_sasl_dir' ? string
+};
+
+type type_qemu_remote = {
+    'remote_display_port_min' ? long
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/libvirtd/pan/schema.pan]: Are these TCP ports? i.e. (1..65535)
Or something else with a restriction?

File: ncm-metaconfig/src/main/metaconfig/libvirtd/pan/schema.pan [hunk 3]
```diff
@@ -99,6 +156,41 @@ type service_sasl2 = {
...
+    include type_qemu_migration
+    'user' ? string
+    'group' ? string
+    'dynamic_ownership' ? boolean
+    'nographics_allow_host_audio' ? boolean
+    'auto_dump_path' ? string
+    'auto_dump_bypass_cache' ? boolean
+    'auto_start_bypass_cache' ? boolean
+    'hugetlbfs_mount' ? string[]
+    'bridge_helper' ? string
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/libvirtd/pan/schema.pan]: This should be an absolute path?

**Comment by alvarosimon** [comment]
@jrha something like this?  I have included `image_format`type within libvirtd schema.

File: ncm-metaconfig/src/main/metaconfig/libvirtd/pan/schema.pan [hunk 4]
```diff
@@ -72,6 +72,72 @@ type type_libvirtd_audit = {
...
+type type_qemu_cgroup = {
+    'cgroup_controllers' ? string[]
+    'cgroup_device_acl' ? string[]
+};
+
+function is_image_format = {
+    image_format = ARGV[0];
+    if(match(image_format, '^(raw|lzop|gzip|bzip2|xz)$')) return(true);
+    error("Bad image format: " + image_format);
+    return(false);
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/libvirtd/pan/schema.pan]: just `false` will do here.

**Comment by jrha** [comment]
Yep, thanks!

**Pull request merged**